libs/designer/src/lib/core/actions/bjsworkflow/add.ts

```diff
@@ -113,6 +113,10 @@ const initializeOperationDetails = async (
     isConnectionRequired = isConnectionRequiredForOperation(manifest);
     connector = manifest.properties?.connector;
 
+    if (connector && connector.id !== operationInfo.connectorId) {
+      dispatch(initializeOperationInfo({ id: nodeId, ...operationInfo, connectorId: connector.id }));
```

Comment:
For certain connectors, connectorId changes based on whether it's being retrieved for search service or if it's retrieved as part of operation manifest. We want the connectorId to be re-initialized to the correct one once it's fetched from operation manifest and is different from the search service one.

Reply:
This is a hack, we cannot have code like this spilled in places just to support some scenario. This looks like a bug in manifest where it does not have correct connectorId which is used to actually fetch the manifest...
You see a manifest is fetched from {connectorId, operationId} tuple.

Please revisit this in the service either backend or search service, it should be a first class connector either from apihub / rp service / hard coded in your client [like we have so many built ins]
If there are PA scenarios which have such connectors, some system needs to be built to support it and not like this in designer but in client service or in backend.

Reply:
Thanks Priti. We have this connector as built in hardcoded in RP but it builds on top of another one and sends different connector id to display data correctly based on when it's being fetched. So, the connector id changes. I agree, this usage of changing connectorId does not seem right so we should look for a way to solve it on our end. I just wanted to run this solution by your team in case it made sense.

I'm thinking another option is to initializeOperationInfo in our host service when we first load the manifest with expanded connector info. Do you think it's okay to change this to just export that function from LAUX instead, or are there be concerns about it too?

Reply:
I think you should have a different operation like apiOperation, manifest and different connectorID, different connection parametera.. and that way in the host when the call comes for getting dynamic data comes you will know which actual connector to call and similarly for connection. 
Many builtins may need specific client codes to handle their cases.

I do have concerns in exposing store update methods to host because it exposes so many entry points in manipulating the data out of what's needed and this can be misused because it is doing things which out of the designed updates... this method is not something which is actually updating something based on user interaction so please don't do it this way and try to create implementation in the calls happening rather trying to fix it in one line.

Reply:
Thank you Priti! Discussed with the team today and we'll try to push the connector team to make it a generic connector and we'll hack the call on our end in the meantime.